qiskit/extensions/unitary.py

```diff
@@ -111,8 +112,9 @@ def _define(self):
         elif self.num_qubits == 2:
             self.definition = two_qubit_cnot_decompose(self.to_matrix()).data
         else:
-            raise NotImplementedError("Not able to generate a subcircuit for "
-                                      "a {}-qubit unitary".format(self.num_qubits))
+            q = QuantumRegister(self.num_qubits, "q")
+            self.definition = [(isometry.Isometry(self.to_matrix(), 0, 0),
+                                [q[x] for x in range(self.num_qubits)], [])]
```

Comment:
a slight simplification
```suggestion
                                q[:], [])]
```